src/role_nextroomer.js

```diff
@@ -308,11 +308,15 @@ roles.nextroomer.settle = function(creep) {
     methods.push(Creep.constructTask);
   }
 
+  // Fill energy after building is done
+  if (creep.room.controller.level >= 4) {
+    methods.push(Creep.transferEnergy);
+  }
```

Comment:
> I think I found a solution that would base filling upon how many parts the creep has.
> If the nextroomer would be only as efficient as a sourcer it has been better in my testing to be building sourcers and having carries deliver to the nextroomers to keep more time on upgrading the controller.
> 
> // Fill energy after building is done
>  if (creep.getActiveBodyparts('work') <= 5 && creep.room.controller.level >= 3) {
>    methods.push(Creep.transferEnergy);
>  }

I like the approach. If you add that you should also make sure that the rooms starts to spawn sourcer.

Maybe you could even make a direct check `nextroomer.work.parts < sourcer.would.have.work.parts`.
This could be done by executing the `role.sourcer.getPartConfig` method and filter the work parts.

https://github.com/TooAngel/screeps/blob/master/src/role_sourcer.js#L22